src/metaImageUtils.h

```diff
@@ -16,15 +16,27 @@
 
 #include "metaImageTypes.h"
 
+#include <memory>
+
 #if (METAIO_USE_NAMESPACE)
 namespace METAIO_NAMESPACE {
 #endif
 
-METAIO_EXPORT bool MET_StringToImageModality(const char * _str,
+METAIO_EXPORT bool MET_StringToImageModality(const std::string _str,
                                       MET_ImageModalityEnumType * _type);
 
 METAIO_EXPORT bool MET_ImageModalityToString(MET_ImageModalityEnumType _type,
-                                      char * _str);
+                                      std::string& _str);
+
+// from https://stackoverflow.com/a/26221725/276168
+template<typename ... Args>
+std::string string_format(const std::string& format, Args ... args)
```

Comment:
Copying code from SO introduces license ambiguities.  Please avoid it.

Reply:
That one was explicitly released under public domain (licensed under CC0 1.0).

Reply:
Based on https://stackoverflow.com/legal/terms-of-service#licensing, the license is https://creativecommons.org/licenses/by-sa/4.0/. 

Instead of adding a reference in the source code, I usually reference the SO post in the commit and mention the commit was adapted from info found in the post.

Reply:
Thanks.  +1 for commit message approach.  The message can also mention the license.